test/utils/asserts.cpp

```diff
@@ -38,4 +38,39 @@ std::ostream& operator<<(std::ostream& os, ExecutionResult result)
     output_result(os, result);
     return os;
 }
+
+namespace test
+{
+std::ostream& operator<<(std::ostream& os, const TypedExecutionResult& result)
+{
+    const auto format_flags = os.flags();
+
+    if (result.has_value)
+    {
+        os << "result(";
+        switch (result.type)
+        {
+        case ValType::i32:
+            os << result.value.as<uint32_t>() << " [0x" << std::hex << result.value.as<uint32_t>()
+               << "] (i32)";
+            break;
+        case ValType::i64:
+            os << result.value.i64 << " [0x" << std::hex << result.value.i64 << "] (i64)";
+            break;
+        case ValType::f32:
+            os << result.value.f32 << " (f32)";
+            break;
+        case ValType::f64:
+            os << result.value.f64 << " (f64)";
+            break;
+        }
+        os << ")";
+    }
+    else
+        os << ExecutionResult{result};
+
+    os.flags(format_flags);
```

Comment:
Ah this is the reset. So is decimal mode on by default according to the std?

Reply:
No. It resets to what state it was before. So explicit `dec` is needed to the first part.